### ISSUE TITLE
[java] UseUtilityClassRule reports false positive with lombok NoArgsConstructor

### ISSUE THREAD

**Issue by clem0110** [pmd/pmd#968]
Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core].

**Affects PMD Version:** 6.0.0

**Rule:** UseUtilityClassRule

**Description:** If you setup a utility factory class using lombok to create the private no arg constructor, a false positive is created that the constructor should be added.

**Code Sample demonstrating the issue:**

```
import lombok.NoArgsConstructor;
import lombok.AccessLevel;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class Foo {
 public static Foo get() {
  return null;
 }
}
```

**Running PMD through:** *[Maven]*

**Comment by clem0110** [comment]
See https://github.com/pmd/pmd/pull/969 for fix
